src/status_checker.c

```diff
@@ -140,6 +239,966 @@ const u8 gDungeonCamouflageTypes[76] = {
+                isTMRecycled = TRUE;
+            }
+        }
+    if ((entityData->heldItem.itemFlags & ITEM_FLAG_EXISTS) && (entityData->heldItem.itemIndex == ITEM_ID_USED_TM)) {
+      xxx_init_itemslot_8090A8C(&entityData->heldItem,entityData->heldItem.numItems + 0x7D,0);
+      isTMRecycled = TRUE;
+    }
+  }
+  if (isTMRecycled) {
+    sub_80522F4(pokemon,target,*gUnknown_80FDC9C); // The Used TM was recharged!
+  }
+  else {
+    sub_80522F4(pokemon,target,*gUnknown_80FDCA0); // But nothing happened!
+  }
+  return isTMRecycled;
+}
+
+bool8 sub_805B7FC(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove *move, s32 param_4)
+{
+    ReflectStatusTarget(pokemon, target);
+    return TRUE;
+}
+
+bool8 sub_805B808(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove *move, s32 param_4)
+{
+    u8 local_20;
+
+    local_20 = 0;
+    sub_806F370(pokemon,target,gUnknown_80F4F7E,1,&local_20,GetMoveType(move),sub_8057600(move,param_4),0,1,0);
+    local_20 = (local_20 == 0);
+    return local_20;
+}
+
+bool8 sub_805B884( struct DungeonEntity * pokemon, struct DungeonEntity * target)
+{ 
+    sub_8077084(pokemon, target, gUnknown_8106A4C, 1);
+    sub_8077910(pokemon, target, 0, TRUE);
+    return TRUE;
+}
+
+bool8 SkullBashMoveAction(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove * move, s32 param_4)
+{
+    bool8 flag;
+
+    if (MoveMatchesChargingStatus(pokemon,move)) {
+        flag = sub_8055640(pokemon,target,move,gUnknown_80F4F58,param_4) != 0 ? 1 : 0;
+        sub_8079764(pokemon);
+    }
+    else {
+        sub_8079618(pokemon,target,6,move,*gUnknown_80FAFF0);
+        flag = TRUE;
+    }
+    return flag;
+}
+
+bool8 sub_805B910(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove * move, s32 param_4)
+{
+    bool8 flag;
+
+    flag = FALSE;
+    if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+        flag = TRUE;
+        if(sub_805727C(pokemon, target, gUnknown_80F4DDC))
+        {
+            sub_8076F80(pokemon, target, gUnknown_8106A50, 1, 1, 0);
+        }
+    }
+    return flag;
+}
+
+bool8 sub_805B968(struct DungeonEntity * pokemon, struct DungeonEntity * target, struct PokemonMove * move, s32 param_4)
+{
+    s32 entityHP;
+    s32 iVar5;
+    bool8 flag;
+
+    flag = FALSE;
+    if (sub_8055640(pokemon,target,move,0x100,param_4) != 0) {
+        flag = TRUE;
+        if (sub_8057308(pokemon,0) != 0) {
+            entityHP = pokemon->entityData->maxHP;
+            if (entityHP < 0) {
+                entityHP = entityHP + 3;
+            }
+            iVar5 = entityHP >> 2;
+            if (iVar5 < 1) {
+                iVar5 = 1;
+            }
+            sub_806F370(pokemon,pokemon,iVar5,0,0,0,sub_8057600(move,param_4),0,1,0);
+        }
+    }
+    return flag;
+}
+
+
+// NOTE: Either Excavate or Rock Smash (90% sure it's Rock Smash)
```

Comment:
Both moves are the same, so chances are that both of them call this function.

Reply:
Yeah, so keep the name then and delete the comment? Or do you prefer something different?

Reply:
Yeah, the comment can be removed.